post/src/main/java/com/oing/dto/dto/PostRankerDTO.java

```diff
@@ -0,0 +1,37 @@
+package com.oing.dto.dto;
+
+
+import lombok.Getter;
+
+@Getter
+public class PostRankerDTO implements Comparable<PostRankerDTO> {
+
+    String memberId;
+    Long postCount;
+    Long commentCount;
+    Long reactionCount;
+
+    public PostRankerDTO(String memberId, Long postCount, Long commentCount, Long reactionCount) {
+        this.memberId = memberId;
+        this.postCount = postCount;
+        this.commentCount = commentCount;
+        this.reactionCount = reactionCount;
+    }
+
+    @Override
+    public int compareTo(PostRankerDTO o) {
+        // 1. postCount 내림차순
+        if (!this.postCount.equals(o.postCount)) return -1 * this.postCount.compareTo(o.postCount);
+        else {
+            // 2. commentCount 내림차순
+            if (!this.commentCount.equals(o.commentCount)) return -1 * this.commentCount.compareTo(o.commentCount);
+            else {
+                // 3. reactionCount 내림차순
+                if (!this.reactionCount.equals(o.reactionCount))
+                    return -1 * this.reactionCount.compareTo(o.reactionCount);
+                    // 4. memberId (가입날짜) 오름차순
+                else return this.memberId.compareTo(o.memberId);
+            }
+        }
+    }
```

Comment:
```suggestion
    public int compareTo(PostRankerDTO o) {
        // 1. postCount 내림차순으로 정렬
        int postCountComparison = Long.compare(o.postCount, this.postCount);
        if (postCountComparison != 0) {
            return postCountComparison;
        }

        // 2. commentCount 내림차순으로 정렬
        int commentCountComparison = Long.compare(o.commentCount, this.commentCount);
        if (commentCountComparison != 0) {
            return commentCountComparison;
        }

        // 3. reactionCount 내림차순으로 정렬
        int reactionCountComparison = Long.compare(o.reactionCount, this.reactionCount);
        if (reactionCountComparison != 0) {
            return reactionCountComparison;
        }

        // 4. memberId (가입날짜) 오름차순으로 정렬
        return this.memberId.compareTo(o.memberId);
    }
```

if-else 문으로 깊게 들어가는 것보다 이렇게 직관적으로 작성하는 건 어떠실까요? (맞는 코드인지는 모르겠네요 😅)

Reply:
와우!  깔끔하네요. 바로 반영했습니다 :>